### PR TITLE
feat(i18n): create a progress indicator

### LINKED ISSUE #50412: Add progress indicators to maps
Here's some mocks:

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #50412

### CODE REVIEW

**Comment by a2937** [comment]
There was an unused variable, but I'm not sure what to do with it. I didn't think it was used before, but without double-checking the code, I'm not sure.

File: client/src/components/Map/index.tsx [hunk 1]
```diff
@@ -87,72 +136,103 @@ function MapLi({
...
+}: MapProps): React.ReactElement {
+  const { t } = useTranslation();
+  let startingIndex = 0;
   return (
     <div className='map-ui' data-test-label='curriculum-map'>
-      <h2 className={forLanding ? 'big-heading' : ''}>
-        {t('landing.core-certs-heading')}
-      </h2>
-      <ul>
-        {coreCurriculum.map((superBlock, i) => (
```
Comment (naomi-lgbt) [inline client/src/components/Map/index.tsx:99]: It was used here.

Comment (a2937) [reply client/src/components/Map/index.tsx:99]: Thank you. That was taken care of.

**Comment by bbsmooth** [comment]
I've got more accessibility issues to add, but it will have to wait just a bit. Here's one to get you started. When you set the view port to 1280px and zoom in 400%, some of the down arrows are out of place, or even completely hidden.
fcc-down-arrow-2
fcc-down-arrow-8

**Comment by a2937** [comment]
I'm still having a little bit of trouble with certain sizes. On smaller screens, it's a little off. However regarding the first issue, I can't figure out how to zoom in all the way like you can right now. I can only zoom in so far before my screen gets completely covered. 
image

**Comment by bbsmooth** [comment]
@a2937 You don't want to use responsive design mode when zooming in. You just want to view the page as normal, set the browser width to 1280px, and then zoom in 400%. Or, using responsive design mode, you could narrow the view port to 320px (that's the equivalent of 1280px at 400%).

**Comment by bbsmooth** [comment]
Sorry, it's taken me a little bit to get to this.

My first comment is that the numbers aren't programmatically associated with the titles. So if I'm navigating through each title with a screen reader, I am only hearing the title and not the step number associated with it. It needs to read something like "Step 1 Responsive Web Design Certification". And that needs to be the accessible name for the link. You can either use `aria-label` to create the proper accessible name or add visually hidden text in the link text. 

Similarly, since we use different looking icons depending on whether you have completed the certification, that information needs to be included in the accessible name as well. So then we are looking at something like "Step 1 Responsive Web Design Certification, Completed" or "Step 2 JavaScript Algorithms and Data Structures (Beta) Certification, Not Completed".

Additionally, the only direct child allowed in a list is a list item (`li`) element. (Well, you can have `script` and `template` elements too, but those wouldn't apply here). So you'll need to replace the outermost `div` wrapping each course with an `li`.

The responsiveness up to 400% is very good now. I don't see the connecting dotted arrows any more, but maybe that's not a deal breaker?

**Comment by Sembauke** [comment]
the tests seem to be failing, please take a look.

**Comment by a2937** [comment]
Okay I didn't realize that the curriculum map had already changed, but it was bound to happen at some point.  I'm going to have update this a good bit to match what's actually online right now. So much of it has changed that it'd probably be easier to use the existing map as a starting point and bring in the good parts of this PR.

**Comment by a2937** [comment]
Going to close this PR and start again with an update base.

**Pull request closed without merging**